packages/resource-deployment/scripts/delete-pools-if-needed.sh

```diff
@@ -0,0 +1,218 @@
+        poolConfigOutdated=true
+    fi
+}
+
+function checkIfPoolConfigOutdated {
+    poolId=$1
+    poolPropertyNamePrefix=$2
+
+    poolConfigOutdated=false
+
+    compareParameterFileToDeployedConfig $poolId "vmSize" "${poolPropertyNamePrefix}VmSize"
+    if [ $poolConfigOutdated == "true" ]; then
+        return
+    fi
+
+    compareParameterFileToDeployedConfig $poolId "maxTasksPerNode" "${poolPropertyNamePrefix}MaxTasksPerNode"
+    if [ $poolConfigOutdated == "true" ]; then
+        return
+    fi
+}
+
+function checkPoolConfigs {
+    for pool in $pools; do
+        camelCasePoolId=$(echo "$pool" | sed -r 's/(-)([a-z])/\U\2/g')
+        checkIfPoolConfigOutdated "$pool" "$camelCasePoolId"
+        if [ $poolConfigOutdated ]; then
+            return
+        fi
+    done
+}
+
+deletePools() {
+    for pool in $pools; do
+        echo "deleting pool $pool"
+        az batch pool delete --account-name $batchAccountName --pool-id $pool --yes
+    done
+
+    for pool in $pools; do
+        waitForDelete $pool
+        echo "Finished deleting $pool"
+    done
+}
+
+waitForDelete() {
+    poolId=$1
+
+    checkIfPoolExists $poolId
+    waiting=false
+    deleteTimeout=1200
+    end=$((SECONDS + $deleteTimeout))
+    while [ "$poolExists" == "true" ] && [ $SECONDS -le $end ]; do
+        if [ "$waiting" != true ]; then
+            waiting=true
+            echo "Waiting for $poolId to delete"
+            printf " - Running .."
+        fi
+
+        sleep 5
+        printf "."
+        checkIfPoolExists $poolId
+    done
+
+    if [ "$poolExists" == "true" ]; then
+        echo "Pool did not finish deleting within $deleteTimeout seconds"
+    fi
+}
+
+checkIfPoolExists() {
+    poolId=$1
+
+    poolExists=$(az batch pool list --account-name "$batchAccountName" --query "[?id=='$poolId']" -o tsv)
+    if [[ -z $poolExists ]]; then
+        poolExists=false
+    else
+        poolExists=true
+    fi
+}
+
+function deletePoolsWhenNodesAreIdle() {
+    command="deletePools"
+    commandName="Delete pool vmss"
+    . "${0%/*}/run-command-when-batch-nodes-are-idle.sh"
+
+    echo "Successfully deleted pools"
+}
+
+function deletePoolsIfNeeded() {
+    az batch account login --name "$batchAccountName" --resource-group "$resourceGroupName"
+    pools=$(az batch pool list --query "[].id" -o tsv)
+    if [[ -z "$pools" ]]; then
+        return
+    fi
+
+    if [[ "$dropPools" == true ]]; then
+        deletePoolsWhenNodesAreIdle
+        return
+    fi
+
+    checkIfVmssAreOld
+    if [[ $areVmssOld == true ]]; then
+        deletePoolsWhenNodesAreIdle
+        return
+    fi
+    
+    checkPoolConfigs
+    if [[ $poolConfigOutdated == true ]]; then
+        deletePoolsWhenNodesAreIdle
+        return
+    fi
+
+    echo "Pools no do not need to be recreated."
+}
+
+# Read script arguments
+while getopts ":r:p:d" option; do
+    case $option in
+    r) resourceGroupName=${OPTARG} ;;
+    p) parameterFilePath=${OPTARG} ;;
+    d) dropPools=true ;;
```

Comment:
should it take param value instead?

Reply:
Right now it's just a flag that doesn't accept parameters, so we set this variable to true if the script is run with the flag enabled. Which is preferred?

Reply:
`d` is part of usage hence should follow the pattern; otherwise just have it as file level var only

Reply:
or set default if no param specified